database.TestCalculateAverage is flaky

It's failed CI here: https://github.com/gpuctl/gpuctl/actions/runs/7856261718/job/21438860855.

And can be reproduced locally:

```
alona@Ashtabula:~/dev/gpuctl$ go test -short ./internal/database/ -count 10
--- FAIL: TestCalculateAverage (0.00s)
    prune_test.go:78: calculateAverage() = {gpu-1 55 30 4500 55 72.5 67.5 1.15 110 1525 1825 725 925 1625140800 [{2 Process2 300} {1 Process1 250}]}, want {gpu-1 55 30 4500 55 72.5 67.5 1.15 110 1525 1825 725 925 1625140800 [{1 Process1 250} {2 Process2 300}]}
FAIL
FAIL    github.com/gpuctl/gpuctl/internal/database      0.017s
FAIL
```

Note: You may need to repeat this command a few times to get it to work, as it's non-deterministic. The `-count` flag is needed to run multiple times, and bypass Go's builtin cacheing


---

I've not investigated the reason, but it smells like rellying on map iteration order.

CC @Euphoride

--- a/internal/database/inmemory.go
+++ b/internal/database/inmemory.go
@@ -1,8 +1,10 @@
 package database
 
 import (
+	"cmp"
 	"errors"
 	"fmt"
+	"slices"
 	"sort"
 	"sync"
 	"time"
@@ -161,6 +163,8 @@ func CalculateAverage(samples []uplink.GPUStatSample) uplink.GPUStatSample {
 
 	var sumMemoryUtil, sumGPUUtil, sumMemoryUsed, sumFanSpeed, sumTemp, sumMemoryTemp, sumGraphicsVoltage, sumPowerDraw, sumGraphicsClock, sumMaxGraphicsClock, sumMemoryClock, sumMaxMemoryClock float64
 	var minTime int64 = samples[0].Time
+
+	// Pid -> Process
 	processesMap := make(map[uint64]uplink.GPUProcInfo)
 
 	for _, sample := range samples {
@@ -182,15 +186,21 @@ func CalculateAverage(samples []uplink.GPUStatSample) uplink.GPUStatSample {
 		}
 
 		for _, proc := range sample.RunningProcesses {
+			// TODO: What if multiple samples have a process with the same PID?
 			processesMap[proc.Pid] = proc
 		}
 	}
 
 	n := float64(len(samples))
+
 	aggregatedProcesses := make([]uplink.GPUProcInfo, 0, len(processesMap))
 	for _, proc := range processesMap {
 		aggregatedProcesses = append(aggregatedProcesses, proc)
 	}
+	// Ensure deterministic order.
+	slices.SortFunc(aggregatedProcesses, func(a, b uplink.GPUProcInfo) int {
+		return cmp.Compare(a.Pid, b.Pid)
+	})
 
 	averagedSample := uplink.GPUStatSample{
 		Uuid:              samples[0].Uuid,

--- a/internal/database/prune_test.go
+++ b/internal/database/prune_test.go
@@ -79,6 +79,57 @@ func TestCalculateAverage(t *testing.T) {
 	}
 }
 
+func TestAverageProcess(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name    string
+		toMerge [][]uplink.GPUProcInfo
+		merged  []uplink.GPUProcInfo
+	}{
+		{"Empty", nil, nil},
+		{"Single Sample",
+			[][]uplink.GPUProcInfo{
+				{{Pid: 1, Name: "foo", MemUsed: 100}, {Pid: 2, Name: "bar", MemUsed: 200}},
+			},
+			[]uplink.GPUProcInfo{{Pid: 1, Name: "foo", MemUsed: 100}, {Pid: 2, Name: "bar", MemUsed: 200}},
+		},
+		{
+			"Three Separate Processes",
+			[][]uplink.GPUProcInfo{
+				{{Pid: 1, Name: "foo", MemUsed: 100}},
+				{{Pid: 2, Name: "bar", MemUsed: 200}},
+				{{Pid: 3, Name: "baz", MemUsed: 300}},
+			},
+			[]uplink.GPUProcInfo{
+				{Pid: 1, Name: "foo", MemUsed: 100},
+				{Pid: 2, Name: "bar", MemUsed: 200},
+				{Pid: 3, Name: "baz", MemUsed: 300},
+			},
+		},
+		// TODO: Test same PID appearing in multiple samples (and decide it's semantics).
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			toMerge := tc.toMerge
+
+			samples := make([]uplink.GPUStatSample, len(toMerge))
+			for i, procs := range tc.toMerge {
+				samples[i].RunningProcesses = procs
+			}
+
+			merged := CalculateAverage(samples).RunningProcesses
+
+			if !reflect.DeepEqual(merged, tc.merged) {
+				t.Errorf("AverageProcesses(%v) = %v, want %v", toMerge, merged, tc.merged)
+			}
+		})
+	}
+
+}
+
 func TestDownsample(t *testing.T) {
 	db := InMemory().(*inMemory)
 

